### PR TITLE
Add issue template

### PR DESCRIPTION
**Pull request by Y--**
First version of issue template - largely inspired (copied) from DuckDB's repo
Fixes https://github.com/duckdb/pg_duckdb/issues/223

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 1]
```diff
@@ -0,0 +1,125 @@
+name: Bug report
+description: Create a report to help us improve
+labels:
+  - needs triage
+body:
+  - type: markdown
+    attributes:
+      value: >
+        Please report security vulnerabilities using GitHub's [report vulnerability form](https://github.com/duckdb/pg_duckdb/security/advisories/new).
```
Comment (wuputah) [inline .github/ISSUE_TEMPLATE/bug_report.yml:9]: seems like we'll need to turn something on here

**Review by wuputah — COMMENTED** [review]
Thanks for starting on this. I think there's no way to preview this, but [looking at duckdb](https://github.com/duckdb/duckdb/issues/new?assignees=&labels=needs+triage&projects=&template=bug_report.yml) this looks quite heavy. For now, I think we want something closer to a template rather than a form - i.e. placeholder text that we can easily modify or delete when opening issues ourselves, rather than something this detailed with separate textareas. I think the form will probably drive us crazy. If we get to the point where we're creating the small minority of issues, this structure might make sense.

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+        Please provide steps to reproduce the behavior, preferably a [minimal reproducible example](https://en.wikipedia.org/wiki/Minimal_reproducible_example). Please adhere the following guidelines:
+
+        * Format the code and the output as [code blocks](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks) using triple backticks:
+
+          ````
+          ```
+          CODE HERE
+          ```
+          ````
+        * Add all required imports for scripts, e.g., `import duckdb`, `import pandas as pd`.
```
Comment (JelteF) [inline .github/ISSUE_TEMPLATE/bug_report.yml:31]: This line can be removed I think.
```suggestion
```

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+        * Remove all prompts from the scripts. This include DuckDB's 'D' prompt and Python's `>>>` prompt. Removing these prompts makes reproduction attempts quicker.
+        * Make sure that the script and its outputs are provided in separate code blocks.
+        * If applicable, please check whether the issue is reproducible via running plain SQL queries from the DuckDB CLI client.
```
Comment (JelteF) [inline .github/ISSUE_TEMPLATE/bug_report.yml:34]: Doesn't really apply I think?
```suggestion
```

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+  - type: input
+    attributes:
+      label: "DuckDB Version:"
+      placeholder: e.g., 22
+    validations:
+      required: true
+  - type: input
+    attributes:
+      label: "Postgres Version:"
+      placeholder: e.g., 16
```
Comment (JelteF) [inline .github/ISSUE_TEMPLATE/bug_report.yml:57]:
```suggestion
      placeholder: e.g., 16.5
```

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+    attributes:
+      label: "OS:"
+      placeholder: e.g., Linux, MacOS, etc.
+      description: Please include operating system version and architecture (e.g., aarch64, x86_64, etc.).
+    validations:
+      required: true
+  - type: input
+    attributes:
+      label: "DuckDB Version:"
+      placeholder: e.g., 22
```
Comment (JelteF) [inline .github/ISSUE_TEMPLATE/bug_report.yml:51]: We ship a fixed version of DuckDB with pg_duckdb, so pg_duckdb version should be enough.
```suggestion
    attributes:
      label: "pg_duckdb Version:"
      placeholder: e.g., 0.1.0
```

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+
+  - type: checkboxes
+    attributes:
+      label: Did you include all code required to reproduce the issue?
+      options:
+        - label: Yes, I have
+
+  - type: checkboxes
+    attributes:
+      label: Did you include all relevant configuration (e.g., CPU architecture, Python version, Linux distribution) to reproduce the issue?
```
Comment (JelteF) [inline .github/ISSUE_TEMPLATE/bug_report.yml:123]:
```suggestion
      label: Did you include all relevant configuration (e.g., CPU architecture, Linux distribution) to reproduce the issue?
```

**Review by JelteF — CHANGES_REQUESTED** [review]

**Comment by JelteF** [comment]
> I think the form will probably drive us crazy.

Totally agreed. I think that's easily solvable by adding a second issue template that we can choose, called something like "Issues reported by maintainers". As long as 90% of users will click the "Bug report" option we'll get the benefit of both.

**Review by wuputah — APPROVED** [review]

**Review by JelteF — APPROVED** [review]

**Pull request merged**